include/opt-sched/Scheduler/lnkd_lst.h

```diff
@@ -176,11 +176,12 @@ class ArenaEntryAllocator
  */
 template <class EntryType>
 std::unique_ptr<EntryAllocator<typename EntryType::value_type>>
-makeDynamicOrArenaAllocator(int MaxSize) {
-  if (MaxSize == INVALID_VALUE)
+makeDynamicOrArenaAllocator(int MaxSize, bool isReadyList = false) {
+  //Logger::Info("list allocator type assigned readylist? %d size? %d", isReadyList, MaxSize);
+  if (MaxSize == INVALID_VALUE && !isReadyList)
     return std::make_unique<DynamicEntryAllocator<EntryType>>();
   else
-    return std::make_unique<ArenaEntryAllocator<EntryType>>(MaxSize);
+    return std::make_unique<ArenaEntryAllocator<EntryType>>(MaxSize == -1 ? 1200 : MaxSize);
```

Comment:
I think the arena allocators are not actually doing what we want them to do. Basically the arena allocator does an allocation per node, and also maintains a free list. We want to allocate a chunk of nodes once, and have node allocation be pulling from a list.

Also, putting an upper limit of 1200 effectively will force a crash if we ever have a correct readylist with more than 1200 elements. While unlikely, this is not an ideal design. The ArenaAllocators need to be reworked.